pepper-apis/src/main/java/org/broadinstitute/ddp/customexport/export/CustomExporter.java

```diff
@@ -0,0 +1,224 @@
+        participantMetaFmt = new ParticipantMetadataFormatter(excludedFields);
+
+
+        List<String> headers = new LinkedList<>(participantMetaFmt.headers());
+
+        Map<String, Integer> activityTagToNormalizedMaxInstanceCounts = new HashMap<>();
+        Map<String, ActivityResponseCollector> responseCollectors = new HashMap<>();
+        Map<String, ActivityAttributesCollector> attributesCollectors = new HashMap<>();
+        for (ActivityExtract activity : activities) {
+            Integer maxInstances = activity.getMaxInstancesSeen();
+            if (maxInstances == null || maxInstances < 1) {
+                LOG.warn("Found max instances count {} for activity tag {}, defaulting to 1", maxInstances, activity.getTag());
+                // NOTE: default to one so we always have one set of columns even if no participant has an instance.
+                maxInstances = 1;
+            }
+
+            List<String> firstFields;
+
+            firstFields = cfg.getStringList(CustomExportConfigFile.FIRST_FIELDS);
+            excludedFields = cfg.getStringList(CustomExportConfigFile.EXCLUDED_ACTIVITY_FIELDS);
+
+            ActivityResponseCollector responseCollector = new ActivityResponseCollector(activity.getDefinition(), firstFields,
+                    excludedFields);
+            ActivityAttributesCollector attributesCollector;
+            attributesCollector = new ActivityAttributesCollector(activity.getAttributesSeen(firstFields, excludedFields));
+
+            activityTagToNormalizedMaxInstanceCounts.put(activity.getTag(), maxInstances);
+            responseCollectors.put(activity.getTag(), responseCollector);
+            attributesCollectors.put(activity.getTag(), attributesCollector);
+
+            for (var i = 1; i <= maxInstances; i++) {
+                headers.addAll(attributesCollector.headers());
+                headers.addAll(responseCollector.getHeaders());
+            }
+        }
+
+        CSVWriter writer = new CSVWriter(output);
+        writer.writeNext(headers.toArray(new String[] {}), false);
+
+        int numWritten = 0;
+        while (participants.hasNext()) {
+            Participant pt = participants.next();
+            List<String> row;
+            try {
+                row = new LinkedList<>(participantMetaFmt.format(pt.getStatus(), pt.getUser()));
+                ComponentDataSupplier supplier = new ComponentDataSupplier(pt.getUser().getAddress(), pt.getProviders());
+                for (ActivityExtract activity : activities) {
+                    String activityTag = activity.getTag();
+                    int maxInstances = activityTagToNormalizedMaxInstanceCounts.get(activityTag);
+                    ActivityResponseCollector responseCollector = responseCollectors.get(activityTag);
+                    ActivityAttributesCollector attributesCollector = attributesCollectors.get(activityTag);
+
+                    List<ActivityResponse> instances = pt.getResponses(activityTag).stream()
```

Comment:
Do we need those 4 checkboxes? If so, those are in the Prequal, so we'll get to grab it from there. Do we need family member data? Those are represented as "child activities", so we'll need to query for the activity definition for those and grab the data from those activity instances.

Also note that I've updated the question stable ids in RGP so they match closer to what we have in DatStat (in the hope that things will look more familiar for study staff). You'll need to update the configuration in Secret Manager.

Reply:
We don't need the 4 checkboxes but we will need the child activities.  Are the nested activities and updated question IDs in develop?

Reply:
Just merged in my PR, so nested activities and updated question IDs are in `develop` now!

Reply:
Nice!